source/compiler-core/slang-json-parser.cpp

```diff
@@ -375,10 +414,45 @@ void JSONWriter::addLexemeValue(JSONTokenType type, const UnownedStringSlice& va
             SLANG_ASSERT(!"Can only emit values");
         }
     }
-    // We have a previous
-    m_state.m_flags |= State::Flag::HasPrevious;
-    // We don't have a key
-    m_state.m_flags &= ~State::Flag::HasKey;
+
+    _postValue();
+}
+
+void JSONWriter::addIntegerValue(int64_t value, SourceLoc loc)
+{
+    _preValue(loc);
+    m_builder << value;
+    _postValue();
+}
+
+void JSONWriter::addFloatValue(double value, SourceLoc loc)
+{
+    _preValue(loc);
+    m_builder << value;
+    _postValue();
+}
```

Comment:
Probably not important right away, but eventually this logic would need to decide how to handle non-finite values, since the JSON spec doesn't specify how they should be encoded. A typical approach is to emit strings like `"inf"`, `"-inf"`, and `"nan"` instead of numeric values.

This goes down an entire rabbit-hole if you need to produce/consume JSON that is compatible with other systems that might have their own conventions (is it `"nan"` or `"NaN"`?), and thus potentially becomes a configuration option on the `JSONWriter` ("here are the literal string values to use when representing inf, -inf, and NaNs").

(There's a whole separate rabbit hole for representing timestamps because there are actual specs/RFCs for that case, IIRC. Hopefully that's not something we need right away.)